ports/stm32/boards/ARDUINO_PORTENTA_H7/mpconfigboard.h

```diff
@@ -120,7 +120,7 @@ void PORTENTA_board_osc_enable(int enable);
 // QSPI flash #1 for storage
 #define MICROPY_HW_QSPI_PRESCALER           (2) // 100MHz
 #define MICROPY_HW_QSPIFLASH_SIZE_BITS_LOG2 (27)
-#define MICROPY_HW_SPIFLASH_SIZE_BITS       (128 * 1024 * 1024)
+#define MICROPY_HW_SPIFLASH_SIZE_BITS       (120 * 1024 * 1024)
```

Comment:
What WiFi firmware would go at this address?

I see the Nicla Vision also uses 120, but the Giga uses 128.  Should they be consistent?

Reply:
@dpgeorge It would also be very nice if we can use the newly added `CYW43_RESOURCE_ATTRIBUTE` feature to add the firmware back to a section, this makes it much easier to relocate it to the QSPI ? For example:
```
#define CYW43_RESOURCE_ATTRIBUTE    __attribute__((aligned(4), section(".cyw43_blob")))
```

Reply:
But then you need a bootloader that's capable of flashing into the QSPI.

Regardless of that, I don't think it's a good idea.  MicroPython will have it's own version of the CYW43 blob included in the firmware, and that may be different to the one stored at the end of the SPI flash.  Best to just leave the one at the end untouched, so it's easy to switch back to Arduino firmware if needed.

Reply:
> But then you need a bootloader that's capable of flashing into the QSPI.

Yes the bootloaders, IDE etc.. all support QSPI, and can flash the blob at the end of the flash separately. The section helps with extracting the firmware blob (removing it from the main firmware image) and then helps to place in the right section in the linker script. For example:
```
# remove the blob from the main image
$(OBJCOPY)  -Obinary -R .big_const* 
```

In linker script I have something like:
```
  .big_const CYW43_MEMORY_OFFSET :
  {
    . = ALIGN(4);
    *(.big_const*)
    . = ALIGN(4);
    _ebig_const = .;
  } > CYW43_MEMORY
```

Reply:
I don't see the benefit of doing it this way.  It will be more hassle for the user to update the WiFi firmware separately.  It's easier if it's just all included in the main firmware.

There's also no guarantee that the WiFi blob in the MicroPython firmware will match what's expected by Arduino / OpenMV.  So in that regard it's better to just leave the last 1MiB of flash alone.

Reply:
> I don't see the benefit of doing it this way

The only benefit of this, at least for us, is that we can reuse the board config files with no or very little changes, which aligns our fork with MicroPython more (an ongoing effort/goal of my mine is to align the fork with upstream as much as possible). But on the other hand the extra section specifications doesn't affect anything else, correct ? Either way this is not something blocking I can fix it in our fork.

Reply:
Just to be clear, you want to relocate the WiFi blobs in the 3 Arduino stm32 boards (in this repo) to the last 1MiB of flash, by using `CYW43_RESOURCE_ATTRIBUTE` is that correct?

If that's the case, I don't think it's a good idea because the layout of the WiFi and BT blobs will be different to the layout expected by the alternative Arduino/OpenMV blobs.

Maybe you can push a commit to show exactly what you mean, then I can understand better.

Reply:
No I just wanted to add the following to the 3 board config files:

```c
#define CYW43_RESOURCE_ATTRIBUTE    __attribute__((aligned(4), section(".cyw43_fw")))
```

This way the firmware blob can be easily manipulated by other build systems (for example: extract it, remove it from main firmware image or relocate it).

Reply:
Oh, OK!  That should be fine then.

Reply:
Done, thanks!